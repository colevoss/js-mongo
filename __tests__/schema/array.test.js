const Schema = require('../../src/Schema');

describe('Schema.array', () => {
  test('Returns valid array', () => {
    const testData = { test: [] };

    expect(Schema.array('test', testData)).toBe(testData.test);
  });

  test('Returns error', () => {
    const testData = { test: 1 };

    const checked = Schema.array('test', testData);

    expect(checked).toBeInstanceOf(TypeError);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.array.__type).toBe('array');
  });
});
