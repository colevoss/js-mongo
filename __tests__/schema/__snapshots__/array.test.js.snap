// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.array Returns error 1`] = `[TypeError: Expected test to be of type array but recieved number of 1]`;
